### PR TITLE
Add Compatibility for PHP 8.2, Drop support for PHP 7.4

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Adds support for PHP 8.2 and fixes a number of psalm issues.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,14 +27,14 @@
     "extra": {
     },
     "require": {
-        "php": "^7.4 || ~8.0.0 || ~8.1.0"
+        "php": "^7.4 || ~8.0.0 || ~8.1.0 || ~8.2.0"
```
Comment (froschdesign) [inline composer.json]: Does this mean we need another patch next month to remove support for 7.4?

Comment (gsteel) [reply composer.json]: There's some ^7.x specific tests here that should also be dropped - it'd be better off in a separate patch.

That said, what's the policy for dropping dead versions of PHP? Is it "As soon as they die", or when the lib wants to use features not available in EOL'd version?

Comment (froschdesign) [reply composer.json]: The question is - should all repositories be touched twice in a short time for these version updates? Has it already been clarified what the best variant would be here?

Comment (gsteel) [reply composer.json]: I don't think this has been clarified… Taken to Slack… 👍

**Review by Ocramius — APPROVED** [review]
LGTM :ship:

**Pull request merged**